Switch to github.com/pkg/errors

poothia opened the pull request:
closes #381

marun commented on pkg/schedulingtypes/plugin.go:
```diff
@@ -204,7 +205,7 @@ func (p *Plugin) ReconcileOverride(qualifiedName util.QualifiedName, result map[
 	overridesMap, err := util.GetOverrides(override)
 	if err != nil {
-		return fmt.Errorf("Error reading cluster overrides for %s %q: %v", p.typeConfig.GetOverride().Kind, qualifiedName, err)
+		return errors.Wrapf(err, "Error reading cluster overrides for %s %q ", p.typeConfig.GetOverride().Kind, qualifiedName)
```
Please remove trailing space

marun commented on test/common/fixtures.go:
```diff
@@ -43,7 +43,7 @@ func typeConfigFixtures() (map[string]*unstructured.Unstructured, error) {
 	files, err := ioutil.ReadDir(path)
 	if err != nil {
-		return nil, fmt.Errorf("Error reading fixture from path %q: %v", path, err)
+		return nil, errors.Wrapf(err, "Error reading fixture from path %q ", path)
```
Remove trailing space

marun commented on test/common/fixtures.go:
```diff
@@ -59,7 +59,7 @@ func typeConfigFixtures() (map[string]*unstructured.Unstructured, error) {
-			return nil, fmt.Errorf("Error reading fixture for %q: %v", typeConfigName, err)
+			return nil, errors.Wrapf(err, "Error reading fixture for %q ", typeConfigName)
 		}
 		fixtures[typeConfigName] = fixture
```
Remove trailing space

marun commented on test/common/testobjects.go:
```diff
@@ -69,7 +69,7 @@ func newTestOverride(apiResource metav1.APIResource, namespace string, clusterNa
 	overridesSlice, ok, err := unstructured.NestedSlice(fixture.Object, "overrides")
 	if err != nil {
-		return nil, fmt.Errorf("Error retrieving overrides field: %v", err)
+		return nil, errors.Wrap(err, "Error retrieving overrides field ")
```
Remove trailing space

marun commented on test/e2e/crd.go:
```diff
@@ -191,7 +192,7 @@ overrides:
 		err = federate.DecodeYAML(strings.NewReader(fixtureYAML), fixture)
 		if err != nil {
-			return nil, nil, nil, fmt.Errorf("Error reading test fixture: %v", err)
+			return nil, nil, nil, errors.Wrap(err, "Error reading test fixture ")
```
Remove trailing space

marun commented on test/e2e/scheduling.go:
```diff
@@ -221,7 +222,7 @@ func createTestObjs(tl common.TestLogger, fedClient clientset.Interface, typeCon
 	fixture, ok := typeConfigFixtures[typeConfigName]
 	if !ok {
-		return "", fmt.Errorf("Unable to find fixture for %q", typeConfigName)
+		return "", errors.New("Unable to find fixture for " + typeConfigName)
```
Should use Errorf instead of New and avoid having to switch to concatenation.

poothia commented:
@marun thanks for reviewing this so thoroughly. I have separated out the vendoring change into its own commit, I Have addressed all your review comments and i'll make a script and add it to ci to flags use of fmt.Errorf too. Thanks.

marun commented on pkg/apis/core/typeconfig/util.go:
```diff
@@ -19,6 +19,7 @@ package typeconfig
 import (
 	"fmt"
 
+	"github.com/pkg/errors"
```
Where no other 3rd party imports are present, please adhere to the following import formatting scheme:

```golang
// system imports

// 3rd party imports (including errors)

// kube imports
```

marun commented on pkg/controller/ingressdns/controller.go:
```diff
@@ -273,19 +273,19 @@ func (c *Controller) getIngressStatusInCluster(cluster, key string) (*corev1.Loa
 		content, err := clusterIngress.MarshalJSON()
 		if err != nil {
-			runtime.HandleError(fmt.Errorf("Failed to marshall the unstructured object: %v", clusterIngress))
+			runtime.HandleError(errors.Errorf("Failed to marshall the unstructured object: %v", clusterIngress))
```
@shashidharatd Why is err not included in the error that is handled?

poothia replied:
@marun @shashidharatd should i include this err?

marun replied:
Please do.

shashidharatd replied:
err is missed. please include

marun commented on pkg/controller/util/deletionhelper/deletion_helper.go:
```diff
@@ -160,9 +160,9 @@ func (dh *DeletionHelper) HandleObjectInUnderlyingClusters(obj runtime.Object, k
-	_, errors := dh.updater.Update(operations)
-	if len(errors) > 0 {
-		return nil, fmt.Errorf("failed to execute deletions for obj %s: %v", objName, errors)
+	_, operrors := dh.updater.Update(operations)
```
Please prefer `lowerCamel` variable names and only use abbreviations that are unambiguous (e.g. `s/errors/operationalErrors/`).

marun commented on pkg/kubefed2/federate/schema.go:
```diff
@@ -19,9 +19,10 @@ package federate
 import (
 	"fmt"
 
+	"github.com/pkg/errors"
```
Please separate this 3rd party import.

marun commented on pkg/kubefed2/federate/util.go:
```diff
@@ -30,6 +30,7 @@ import (
 	"k8s.io/client-go/rest"
 
 	"github.com/kubernetes-sigs/federation-v2/pkg/apis/core/typeconfig"
+	"github.com/pkg/errors"
```
Please separate this 3rd party import.

marun commented on test/common/fixtures.go:
```diff
@@ -17,12 +17,12 @@ limitations under the License.
 	"strings"
 
 	"github.com/kubernetes-sigs/federation-v2/pkg/kubefed2/federate"
+	"github.com/pkg/errors"
```
Please separate this 3rd party import.

marun commented on test/common/testobjects.go:
```diff
@@ -17,12 +17,12 @@ limitations under the License.
 
 	"github.com/kubernetes-sigs/federation-v2/pkg/apis/core/typeconfig"
 	"github.com/kubernetes-sigs/federation-v2/pkg/controller/util"
+	"github.com/pkg/errors"
```
Please separate this 3rd party import.

marun commented on test/e2e/scheduling.go:
```diff
@@ -35,6 +35,7 @@ import (
 	"github.com/kubernetes-sigs/federation-v2/test/common"
 	"github.com/kubernetes-sigs/federation-v2/test/e2e/framework"
 	"github.com/kubernetes-sigs/federation-v2/test/e2e/framework/managed"
+	"github.com/pkg/errors"
```
Please separate this 3rd party import

poothia commented on pkg/controller/dnsendpoint/controller.go:
```diff
@@ -188,7 +188,7 @@ func (d *controller) processItem(key string) error {
 	obj, exists, err := d.dnsObjectStore.GetByKey(key)
 	if err != nil {
-		return fmt.Errorf("error fetching object with key %s from store: %v", key, err)
+		return errors.Wrapf(err, "error fetching object with key %s from store", err)
```
@shashidharatd thanks for reviewing this, I have fixed it. i am sorry i missed that. Thanks.

shashidharatd commented:
@poothia, request to rebase. conflicts happened after #548 merged. Thanks

shashidharatd commented:
/lgtm
Thanks @poothia 
@marun, request to approve.
